### PR TITLE
Allow overriding Content-Type

### LINKED ISSUE #3194: Server does not set 'charset` in 'Content-Type' response header
The server response doesn't set `charset` in `content-type` header. This might be a limitation that some website needs some some special encoding.

- [x] I have searched the [issues](https://github.com/zeit/next.js/issues) of this repository and believe that this is not a duplicate.

## Expected Behavior
By default `content-type` response header should be `text/html; charset=utf-8`
Also there should also be an option to override the default charset in `next.config.js`
```js
// next.config.js
module.exports = {
  encoding: 'iso-8859-1'
}
```
## Current Behavior
The `content-type` is hard-coded as `text/html`

### PR DESCRIPTION
**Pull request by timneutkens**
Fixes #3194

Closes #3185

### CODE REVIEW

**Review by sergiodxa — APPROVED** [review]

**Comment by liweinan0423** [comment]
@timneutkens thanks for fixing the issue. There might be another concern - if setting `content-type` in `getInitialProps`, would static exporting still work because of `res` is used in the component

**Comment by howks** [comment]
Why default is still `text/htm` in sendHTML function? It should be `text/html; charset=utf-8`.
With this change, 99% of the time, you will not need to overwrite the Content-Type in getInitialProps.

File: server/render.js [hunk 1]
```diff
@@ -185,7 +185,9 @@ export function sendHTML (req, res, html, method, { dev }) {
   }
 
   res.setHeader('ETag', etag)
-  res.setHeader('Content-Type', 'text/html')
+  if (!res.getHeader('Content-Type')) {
+    res.setHeader('Content-Type', 'text/html')
```
Comment (howks) [inline server/render.js]: Default should be `text/html; charset=utf-8`

**Comment by timneutkens** [comment]
@howks fair enought. Feel free to send a PR!

**Pull request merged**